File: RadarHub.Back/RadarHub.API/Controllers/PoderController.cs
using Microsoft.AspNetCore.Mvc;
using RadarHub.Dominio.Entidades;
using RadarHub.Dominio.Servicos;
using RSK.API.Controllers;
using RSK.Dominio.Interfaces;
using RSK.Dominio.Notificacoes.Interfaces;
using RSK.Dominio.Servicos;

namespace RadarHub.API.Controllers
{
    [ApiController]
    [Route("api/[controller]")]
    public class PoderController : ApiConsultaControllerBase<Poder>
    {
        private readonly PoderServico _poderServico;
        private readonly INotificador _mensagens;

        public PoderController(IServicoConsultaBase<Poder> servicoConsulta, PoderServico poderServico, INotificador mensagens) : base(servicoConsulta)
        {
            _poderServico = poderServico;
            _mensagens = mensagens;
        }


        [HttpPost("Importar")]
        public async Task<IActionResult> Importar()
        {
            var sucesso = await _poderServico.ImportarAsync();
            var resultado = _mensagens.ObterMensagens();

            if (!sucesso)
            {
                return BadRequest(resultado);
            }

            return Ok(resultado);
        }
    }
}

File: RadarHub.Web/src/servicos/PoderController.ts
import { Poder } from "@/DTO/Poder";
import { ConsultaServicoBase } from "./ConsultaServicoBase";
import http from "./http-comon";

class PoderController extends ConsultaServicoBase<Poder> {
    constructor() {
        super("Poder");
    }

    public async importar() {
        return http.post(`/poder/importar`);
    }
}

const PoderServico = new PoderController()

export default PoderServico;